src/status_im/network/net_info.cljs

```diff
@@ -7,12 +7,14 @@
             [status-im.mailserver.core :as mailserver]
             [status-im.chaos-mode.core :as chaos-mode]
             [status-im.native-module.core :as status]
-            [status-im.ui.screens.mobile-network-settings.events :as mobile-network]))
+            [status-im.ui.screens.mobile-network-settings.events :as mobile-network]
+            [status-im.wallet.core :as wallet]))
 
 (fx/defn change-network-status
   [{:keys [db] :as cofx} is-connected?]
   (fx/merge cofx
             {:db (assoc db :network-status (if is-connected? :online :offline))}
+            (if is-connected? (wallet/update-balances nil))
```

Comment:
You can use when here instead of if.

Also subscriptions are for views, in events you get the values in db so you can check the balances from 'db' here, check how the balance is checked in the subscription to get it from db. I'm on mobile but if you need more details I can check and point you to the right place